### PR TITLE
feat: Transactions imports

### PR DESCRIPTION
**Pull request by Betree**
For https://github.com/opencollective/opencollective/issues/7416
Frontend: https://github.com/opencollective/opencollective-frontend/pull/10466

### CODE REVIEW

File: server/graphql/v2/mutation/TransactionImportsMutations.ts [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        type: new GraphQLNonNull(GraphQLAccountReferenceInput),
+        description: 'Account that will hold the import (usually the host)',
+      },
+      source: {
+        type: new GraphQLNonNull(GraphQLNonEmptyString),
+        description: 'Source of the import (e.g. "Bank of America", "Eventbrite", etc...)',
+      },
+      name: {
+        type: new GraphQLNonNull(GraphQLNonEmptyString),
+        description: 'Name of the import (e.g. "Contributions May 2021", "Tickets for Mautic COnference 2024")',
```
Comment (znarf) [inline server/graphql/v2/mutation/TransactionImportsMutations.ts:32]:
```suggestion
        description: 'Name of the import (e.g. "Contributions May 2021", "Tickets for Mautic Conference 2024")',
```

File: server/graphql/v2/mutation/TransactionImportsMutations.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+          name: args.name,
+          type: args.type,
+        },
+        {
+          UserTokenId: req.userToken?.id,
+        },
+      );
+    },
+  },
+  importTransactions: {
```
Comment (znarf) [inline server/graphql/v2/mutation/TransactionImportsMutations.ts:63]: description for the mutation?

File: server/graphql/v2/mutation/TransactionImportsMutations.ts [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+import { idDecode } from '../identifiers';
+import { fetchAccountWithReference, GraphQLAccountReferenceInput } from '../input/AccountReferenceInput';
+import { getValueInCentsFromAmountInput } from '../input/AmountInput';
+import { GraphQLTransactionsImportRowCreateInput } from '../input/TransactionsImportRowCreateInput';
+import { GraphQLTransactionsImportRowUpdateInput } from '../input/TransactionsImportRowUpdateInput';
+import { GraphQLTransactionsImport } from '../object/TransactionsImport';
+
+const transactionImportsMutations = {
+  createTransactionsImport: {
+    type: new GraphQLNonNull(GraphQLTransactionsImport),
```
Comment (znarf) [inline server/graphql/v2/mutation/TransactionImportsMutations.ts:20]: description for the mutation?

File: server/graphql/v2/object/Host.js [hunk 4]
```diff
@@ -1742,6 +1743,55 @@ export const GraphQLHost = new GraphQLObjectType({
...
+        args: {
+          ...CollectionArgs,
+          orderBy: {
+            type: new GraphQLNonNull(GraphQLChronologicalOrderInput),
+            description: 'The order of results',
+            defaultValue: CHRONOLOGICAL_ORDER_INPUT_DEFAULT_VALUE,
+          },
+        },
+        async resolve(host, args, req) {
+          checkRemoteUserCanUseHost(req);
```
Comment (znarf) [inline server/graphql/v2/object/Host.js:1758]: Consistency with checkRemoteUserCanUseTransactions ?

File: server/graphql/v2/mutation/TransactionImportsMutations.ts [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+        type: new GraphQLNonNull(GraphQLNonEmptyString),
+        description: 'ID of the import to add transactions to',
+      },
+      rows: {
+        type: new GraphQLNonNull(new GraphQLList(new GraphQLNonNull(GraphQLTransactionsImportRowUpdateInput))),
+        description: 'Rows to update',
+      },
+    },
+    resolve: async (_: void, args, req: Request) => {
+      checkRemoteUserCanUseTransactions(req);
```
Comment (znarf) [inline server/graphql/v2/mutation/TransactionImportsMutations.ts:132]: consistency with checkRemoteUserCanUseHost ?

File: server/graphql/v2/object/TransactionsImport.ts [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+        }
+      },
+    },
+    source: {
+      type: new GraphQLNonNull(GraphQLNonEmptyString),
+      description: 'Source of the import (e.g. "Bank of America", "Eventbrite", etc...)',
+    },
+    name: {
+      type: new GraphQLNonNull(GraphQLNonEmptyString),
+      description: 'Name of the import (e.g. "Contributions May 2021", "Tickets for Mautic COnference 2024")',
```
Comment (znarf) [inline server/graphql/v2/object/TransactionsImport.ts:41]:
```suggestion
      description: 'Name of the import (e.g. "Contributions May 2021", "Tickets for Mautic Conference 2024")',
```

**Review by znarf — CHANGES_REQUESTED** [review]
Looks good!

Some description of the mutations would be appreciated to make it easier to understand what their purpose. 

In term of scope check, maybe have another look of we want the scope "transactions" or the scope "host" or both. I ultimately think it needs to be consistent for the whole feature.

**Comment by Betree** [comment]
@znarf Thanks for the review! I've updated the GraphQL documentation and consolidated scope checks under `checkRemoteUserCanUseTransactions`.

**Review by znarf — APPROVED** [review]
Thank you!

**Pull request merged**